Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm

```diff
@@ -201,23 +201,26 @@
     if (!textIndicatorData)
         return;
 
-    m_webPage->addTextAnimationForAnimationID(initialAnimationUUID, { TextAnimationType::Initial, WTF::UUID(WTF::UUID::emptyValue) }, *textIndicatorData);
+    m_webPage->addTextAnimationForAnimationID(initialAnimationUUID, { TextAnimationType::Initial, WebCore::WritingTools::RunAnimation::RunAnimation, WTF::UUID(WTF::UUID::emptyValue) }, *textIndicatorData);
 
     m_initialAnimations.set(sessionUUID, initialAnimationUUID);
 }
 
 
-void TextAnimationController::addSourceTextAnimation(const WTF::UUID& sessionUUID, const WebCore::CharacterRange& replacingRange, const String string, WTF::CompletionHandler<void(void)>&& completionHandler)
+void TextAnimationController::addSourceTextAnimation(const WTF::UUID& sessionUUID, const WebCore::CharacterRange& replacingRange, const String string, WTF::CompletionHandler<void(WebCore::WritingTools::RunAnimation)>&& completionHandler)
 {
 #if PLATFORM(MAC)
     auto previouslyReplacedRange = m_alreadyReplacedRanges.getOptional(sessionUUID);
     auto replaceCharacterRange = replacingRange;
 
+    WebCore::WritingTools::RunAnimation runAnimation = WebCore::WritingTools::RunAnimation::RunAnimation;
     if (previouslyReplacedRange) {
-        if ((*previouslyReplacedRange)->range == replacingRange && (*previouslyReplacedRange)->string == string)
-            return;
-
-        replaceCharacterRange = remainingCharacterRange(replacingRange, (*previouslyReplacedRange)->range);
+        // If the text is the same as has been repalced before, this is the final replace, we we shouldn't
```

Comment:
> If the text is the same as has been repalced before, this is the final replace

this isn't _strictly_ true; I think it's technically valid for this to happen even if it has yet to finish